libnetwork/network/cni_interface.go

```diff
@@ -0,0 +1,121 @@
+//go:build (linux || freebsd) && cni
```

Comment:
Nit: I'd prefer something either like interface_cni.go and interface_cni_unsupported.go, or cni.go and cni_unsupported.go to be consistent with our other file naming patterns in our repos.